packages/expo-asset/android/src/main/java/expo/modules/asset/AssetModule.kt

```diff
@@ -0,0 +1,101 @@
+package expo.modules.asset
+
+import android.net.Uri
+import expo.modules.kotlin.Promise
+import expo.modules.interfaces.filesystem.Permission
+import expo.modules.kotlin.AppContext
+import expo.modules.kotlin.exception.CodedException
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+import java.io.File
+import java.io.FileInputStream
+import java.net.URI
+import java.security.DigestInputStream
+import java.security.MessageDigest
+
+internal class UnableToDownloadAssetException(url: String) :
+  CodedException("Unable to download asset from url: $url")
+
+class AssetModule : Module() {
+  private fun getMD5HashOfFilePath(uri: URI): String {
+    val md = MessageDigest.getInstance("MD5")
+    return md.digest(uri.toString().toByteArray()).joinToString("") { "%02x".format(it) }
+  }
+
+  private fun getMD5HashOfFileContent(file: File): String? {
+    try {
+      val md = MessageDigest.getInstance("MD5")
+      val fis = FileInputStream(file)
+      val dis = DigestInputStream(fis, md)
+
+      while (dis.read() != -1) {
+        // Read the file content
+      }
+
+      val digest = md.digest()
+      fis.close()
+
+      val result = StringBuilder()
+      for (b in digest) {
+        result.append(String.format("%02x", b))
+      }
+      return result.toString()
```

Comment:
```suggestion
      return DigestInputStream(fis, md).use { dis ->
  	    while (dis.read() != -1) {
	        // Read the file content
	      }
	
	      md.digest().joinToString(separator = "") { "%02x".format(it) }
      }
```
It's safer to close streams with the `use` function. Also, if you close the `DigestInputStream` it should automatically close the `FileInputStream`, which was used during construction.